Extensions/timestamps.js

```diff
@@ -224,12 +245,91 @@ XKit.extensions.timestamps = new Object({
 
 			var timestamp = responseData.posts[0].timestamp;
 			date_element.html(this.format_date(moment(new Date(timestamp * 1000))));
-			date_element.removeClass("xtimestamp_loading");
+			date_element.removeClass("xtimestamp-loading");
 			XKit.storage.set("timestamps", "xkit_timestamp_cache_" + post_id, timestamp);
 		})
 		.catch(() => this.show_failed(date_element));
 	},
 
+	react_add_timestamps: function() {
+		var posts = $("[data-id]:not(.xkit_timestamps)");
+		posts.addClass("xkit_timestamps");
+
+		if (posts.length === 0) {
+			return;
+		}
+
+		XKit.extensions.timestamps.check_quota();
+
+		posts.each(function() {
+			var post = $(this);
+
+			var post_id = post.attr("data-id");
+
+			var xtimestamp_class = "xtimestamp";
+			if (XKit.extensions.timestamps.in_search) {
+				xtimestamp_class = "xtimestamp-in-search";
+			}
+
+			var xtimestamp_html = `<div class="xkit_timestamp_${post_id} ${xtimestamp_class} xtimestamp-loading">&nbsp;</div>`;
+			$(xtimestamp_html).insertAfter(post.find("header"));
+
+			var note = $(".xkit_timestamp_" + post_id);
+			XKit.extensions.timestamps.react_fetch_timestamp(post_id, note);
+		});
+	},
+
+	react_add_reblog_timestamps: function() {
+		var reblogs = $(XKit.extensions.timestamps.reblogs_class).not(".xkit_timestamps");
+
+		if (XKit.extensions.timestamps.preferences.reblogs.value === "op") {
+			var posts = $("[data-id]");
+			reblogs = posts.map(function() { return $(this).children(XKit.extensions.timestamps.reblogs_class).not(".xkit_timestamps").get(0); });
+		}
+
+		reblogs
+		.addClass("xkit_timestamps")
+		.each(function() {
+			var reblog = $(this);
+
+			if (reblog.length === 0) { return; }
+
+			try {
+				var post_id = reblog.find(XKit.extensions.timestamps.blog_link_class)[1].href.split("/").slice(-2)[0];
+				var blog_name = reblog.find(XKit.extensions.timestamps.blog_link_class)[1].href.split("/")[2].split(".")[0];
+			} catch (e) {
+				reblog.find(XKit.extensions.timestamps.reblog_headers_class).append(`<div class="xtimestamp xtimestamp-reblog">&nbsp;</div>`);
+				note = reblog.find(".xtimestamp.xtimestamp-reblog");
+				XKit.extensions.timestamps.show_failed(note);
+				return;
+			}
+
+			var normal_html = `<div class="xkit_timestamp_${post_id} xtimestamp xtimestamp-reblog xtimestamp-loading">&nbsp;</div>`;
+			reblog.find(XKit.extensions.timestamps.reblog_headers_class).append(normal_html);
+			var note = $(`.xkit_timestamp_${post_id}`);
+
+			// XKit.interface.react.post_props doesn't appear to work for reblogs yet
+//			XKit.extensions.timestamps.react_fetch_timestamp(post_id, note);
```

Comment:
unnecessary comments; `post_props` will never have reblog timestamps, since they aren't saved as part of the trail

```suggestion
```

Reply:
):